development/tools/pkg/release/github.go

```diff
@@ -0,0 +1,64 @@
+package release
+
+import (
+	"context"
+	"os"
+
+	"golang.org/x/oauth2"
+
+	"github.com/google/go-github/github"
+	"github.com/kyma-project/test-infra/development/tools/pkg/common"
+)
+
+// GithubAPI exposes functions to interact with Github releases
+type GithubAPI interface {
+	CreateGithubRelease(ctx context.Context, releaseVersion string, releaseBody string, targetCommit string, isPreRelease bool) (*github.RepositoryRelease, *github.Response, error)
```

Comment:
So many strings in one function. It's easy to make a mistake. Have you thought about wrapping it i a structure, like `ReleaseOptions` and passing it as one argument.
Pro: keywords are used on call-site.
Con: Not all arguments are mandatory.
Just something to consider, I'm not forcing it.